synapse/http/client.py

```diff
@@ -42,27 +42,61 @@
 
 from synapse.api.errors import Codes, HttpResponseException, SynapseError
 from synapse.http import cancelled_to_request_timed_out_error, redact_uri
-from synapse.http.endpoint import SpiderEndpoint
 from synapse.util.async_helpers import timeout_deferred
 from synapse.util.caches import CACHE_SIZE_FACTOR
 from synapse.util.logcontext import make_deferred_yieldable
 
 logger = logging.getLogger(__name__)
 
 outgoing_requests_counter = Counter("synapse_http_client_requests", "", ["method"])
-incoming_responses_counter = Counter("synapse_http_client_responses", "",
-                                     ["method", "code"])
+incoming_responses_counter = Counter(
+    "synapse_http_client_responses", "", ["method", "code"]
+)
+
+
+@implementer(IResolutionReceiver)
+class IPListResolutionReceiver(object):
```

Comment:
I know it's internal, but I think a very quick docstring would be useful on this. What does the deferred returned by `get_deferred` resolve to? (And assuming it doesn't follow the synapse logcontext rules, please make this clear)